src/Test/L0/Worker/Build/TrackingManagerL0.cs

```diff
@@ -0,0 +1,524 @@
+        {
+            using (TestHostContext hc = Setup())
+            {
+                // Act.
+                TrackingConfigBase config = _trackingManager.LoadIfExists(
+                    _ec.Object,
+                    Path.Combine(_workFolder, "foo.json"));
+
+                // Assert.
+                Assert.Null(config);
+            }
+        }
+
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Worker")]
+        public void MarksTrackingConfigForGarbageCollection()
+        {
+            using (TestHostContext hc = Setup())
+            {
+                // Arrange.
+                // It doesn't matter for this test whether the line endings are CRLF or just LF.
+                const string TrackingContents = @"{
+  ""build_artifactstagingdirectory"": ""b00335b6\\a"",
+  ""agent_builddirectory"": ""b00335b6"",
+  ""collectionUrl"": ""http://contoso:8080/tfs/DefaultCollection/"",
+  ""definitionName"": ""M87_PrintEnvVars"",
+  ""fileFormatVersion"": 3,
+  ""lastRunOn"": ""09/16/2015 23:56:46 -04:00"",
+  ""build_sourcesdirectory"": ""b00335b6\\gitTest"",
+  ""common_testresultsdirectory"": ""b00335b6\\TestResults"",
+  ""collectionId"": ""7aee6dde-6381-4098-93e7-50a8264cf066"",
+  ""definitionId"": ""7"",
+  ""hashKey"": ""b00335b6923adfa64f46f3abb7da1cdc0d9bae6c"",
+  ""repositoryUrl"": ""http://contoso:8080/tfs/DefaultCollection/_git/gitTest"",
+  ""system"": ""build""
+}";
+                Directory.CreateDirectory(_workFolder);
+                string trackingFile = Path.Combine(_workFolder, "trackingconfig.json");
+                File.WriteAllText(trackingFile, TrackingContents);
+                TrackingConfig config = _trackingManager.LoadIfExists(_ec.Object, trackingFile) as TrackingConfig;
+                Assert.NotNull(config);
+
+                // Act.
+                _trackingManager.MarkForGarbageCollection(_ec.Object, config);
+
+                // Assert.
+                string gcDirectory = Path.Combine(
+                    _workFolder,
+                    Constants.Build.Path.SourceRootMappingDirectory,
+                    Constants.Build.Path.GarbageCollectionDirectory);
+                Assert.True(Directory.Exists(gcDirectory));
+                string[] gcFiles = Directory.GetFiles(gcDirectory);
+                Assert.Equal(1, gcFiles.Length);
+                string gcFile = gcFiles.Single();
+                string gcContents = File.ReadAllText(gcFile);
+                Assert.Equal(TrackingContents, gcContents);
+                // File name should a GUID.
+                Assert.True(Regex.IsMatch(Path.GetFileNameWithoutExtension(gcFile), "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$"));
+                // File name should not be the default GUID.
+                Assert.NotEqual("00000000-0000-0000-0000-000000000000", Path.GetFileNameWithoutExtension(gcFile));
+            }
+        }
+
+#if OS_WINDOWS
```

Comment:
An assertion failed from this test failed on Linux. Technically none of the Legacy format tests are required on xplat, but this is the only one that I added an `#if` around.

The reason it failed on Linux is because of the test data I'm using. The segments of the Windows-style legacy source folder path doesn't get split apart correctly by Path.GetDirectoryName() on Linux.

The other legacy format tests passed, so I just let them run on Linux/OSX.